src/heretic/p_map.c

```diff
@@ -1199,14 +1199,20 @@ boolean PTR_AimTraverse(intercept_t * in)
 
         dist = FixedMul(attackrange, in->frac);
 
-        if (li->frontsector->floorheight != li->backsector->floorheight)
+        // Added checks if there is no backsector to prevent crashing.
+        // Crashes didn't happen in the DOS version of Heretic, assuming 
```

Comment:
No need to assume. I tested and the original DOS4G(W) extender mapped conventional memory to low addresses. Reading NULL or around there would have read from the interrupt vector table and gotten random (but reasonably consistent for a single run) bytes.

Reply:
Something like this, maybe?
```
        // Added checks if there is no backsector to prevent crashing.
        // Crashes didn't happen in the DOS version of Heretic because of
        // reading random bytes from memory, consistent enough for a single
        // run of conditions below.
        // See https://github.com/chocolate-doom/chocolate-doom/issues/1665
```

Reply:
Specifically reading from NULL pointer works, not any random pointer.